src/app/components/POCs/CamelDSLCodeEditor/CamelDSLCodeEditor.tsx

```diff
@@ -0,0 +1,131 @@
+import React, {
+  useCallback,
+  useEffect,
+  useRef,
+  VoidFunctionComponent,
+} from "react";
+import * as monacoEditor from "monaco-editor";
+import MonacoEditor from "./camelDSLutils";
+import "./CamelDSLCodeEditor.css";
+
+export interface CamelDSLCodeEditorProps {
+  /** The initial content of the code editor */
+  code: string;
+  /** Callback for changes in the code */
+  onChange: (value: string) => void;
+  /** Width of the code editor */
+  width?: string | number;
+  /** Height of the code editor */
+  height?: string | number;
+  /** Sink names to use as suggested values for the 'to' property */
+  sinkConnectorsNames: string[];
+}
+
+interface Range {
+  startLineNumber: number;
+  endLineNumber: number;
+  startColumn: number;
+  endColumn: number;
+}
+
+interface Suggestion {
+  label: string;
+  kind: number;
+  documentation: string;
+  insertText: string;
+  range: Range;
+}
+
+const CamelDSLCodeEditor: VoidFunctionComponent<CamelDSLCodeEditorProps> = (
+  props
+) => {
+  const {
+    code,
+    onChange,
+    width = "100%",
+    height = "500",
+    sinkConnectorsNames,
+  } = props;
+
+  const completionProvider = useRef<monacoEditor.IDisposable>();
+
+  const createToProposals = useCallback(
+    (range: Range): Suggestion[] => {
+      return (sinkConnectorsNames ?? []).map((name) => ({
+        label: `"${name}"`,
+        kind: monacoEditor.languages.CompletionItemKind.Function,
+        documentation: "This is a sink I created before.",
+        insertText: `"${name}"`,
+        range: range,
+      }));
```

Comment:
This is where `to:` suggestions are prepared. 
You can see them by hitting `ctrl+space` next to the `to:` property in the correspondent story.
I've left the `documentation` property there with some random text because I'm not sure how we could use it. We could eventually show the sink type or some other meta data.